### PR TITLE
feat: setEmergencyExit should allow to be called by vault guardian and vault management

### LINKED ISSUE #294: BaseStrategy setEmergencyExit should allow to be called by vault guardian and vault management
`BaseStrategy` `setEmergencyExit()` should allow to be called by vault guardian and vault management

### PR DESCRIPTION
**Pull request by steffenix**
fixes #294

### CODE REVIEW

**Review by fubuloubu — APPROVED** [review]

File: contracts/BaseStrategy.sol [hunk 1]
```diff
@@ -273,6 +273,10 @@ abstract contract BaseStrategy {
         _;
     }
 
+    modifier onlyFireFighters() {
+        require(msg.sender == strategist || msg.sender == governance() || msg.sender == vault.guardian() || vault.management(), "!authorized");
```
Comment (fubuloubu) [inline contracts/BaseStrategy.sol:277]: Mistake
```suggestion
        require(msg.sender == strategist || msg.sender == governance() || msg.sender == vault.guardian() || msg.sender == vault.management(), "!authorized");
```

File: contracts/BaseStrategy.sol [hunk 2]
```diff
@@ -273,6 +273,14 @@ abstract contract BaseStrategy {
         _;
     }
 
+    modifier onlyFireFighters() {
```
Comment (fubuloubu) [inline contracts/BaseStrategy.sol:276]:
```suggestion
    modifier onlyEmergencyAuthorized() {
```

**Review by fubuloubu — APPROVED** [review]

File: contracts/BaseStrategy.sol (hunk 2 above)
Comment (steffenix) [reply contracts/BaseStrategy.sol:276]: updated.

**Comment by fubuloubu** [comment]
Do we want this against `develop`?

**Comment by steffenix** [comment]
> Do we want this against `develop`?

Master and develop

**Comment by fubuloubu** [comment]
> > Do we want this against `develop`?
> 
> Master and develop

But do we want against `develop` first, and then upstream merge to `master` like last time? This is for an audit comment, right?

**Comment by steffenix** [comment]
> > > Do we want this against `develop`?
> > 
> > 
> > Master and develop
> 
> But do we want against `develop` first, and then upstream merge to `master` like last time? This is for an audit comment, right?

That's from a request from @poolpitako 
https://github.com/yearn/yearn-vaults/issues/294

**Pull request merged**